src/Model/CompositeTypeGo.cs

```diff
@@ -206,56 +188,85 @@ public string AddHTTPResponse()
+                indented.Append(((CompositeTypeGo)BaseModelType).Fields(forMarshaller));
             }
 
             // Emit each property, except for named Enumerated types, as a pointer to the type
             foreach (var property in properties)
             {
+                if (!forMarshaller && !string.IsNullOrEmpty(property.Documentation))
+                {
+                    indented.AppendFormat("// {0} - {1}\n", property.Name, property.Documentation);
+                }
                 var enumType = property.ModelType as EnumTypeGo;
                 if (enumType != null && enumType.IsNamed)
                 {
                     indented.AppendFormat("{0} {1} {2}\n",
                                     property.Name,
                                     enumType.Name,
-                                    property.JsonTag());
+                                    property.Tag());
 
                 }
                 else if (property.ModelType is DictionaryType)
                 {
-                    indented.AppendFormat("{0} *{1} {2}\n", property.Name, (property.ModelType as DictionaryTypeGo).Name, property.JsonTag());
+                    var typeName = (property.ModelType as DictionaryTypeGo).Name;
+                    if (property.IsMetadata)
+                    {
+                        // use custom type instead of a map[string]string
+                        typeName = "Metadata";
+                    }
+                    indented.AppendFormat("{0} {1} {2}\n", property.Name, typeName, property.Tag());
                 }
-                else if (property.ModelType.PrimaryType(KnownPrimaryType.Object))
+                else if (property.ModelType.IsPrimaryType(KnownPrimaryType.Object))
                 {
                     // TODO: I don't think this is the best way to handle object types
-                    indented.AppendFormat("{0} *{1} {2}\n", property.Name, property.ModelType.Name, property.JsonTag());
+                    indented.AppendFormat("{0} *{1} {2}\n", property.Name, property.ModelType.Name, property.Tag());
                 }
                 else if (property.ShouldBeFlattened())
                 {
                     // embed as an anonymous struct.  note that the ordering of this clause is
                     // important, i.e. we don't want to flatten primary types like dictionaries.
-                    indented.AppendFormat("*{0} {1}\n", property.ModelType.Name, property.JsonTag());
+                    indented.AppendFormat("*{0} {1}\n", property.ModelType.Name, property.Tag());
                     property.Extensions[SwaggerExtensions.FlattenOriginalTypeName] = Name;
                 }
+                else if (!string.IsNullOrEmpty(NextLink) && property.Name.EqualsIgnoreCase(NextLink))
+                {
+                    // use custom type instead of *string
+                    indented.Append($"{NextLink} Marker `xml:\"{NextLink}\"`");
+                }

```

Comment:
Could there be a case in which this is generated for an API that uses json?

Reply:
Good catch will fix.